src/Model/ModelManagerInterface.php

```diff
@@ -97,135 +102,145 @@ public function createQuery($class, $alias = 'o');
      * composed of multiple columns. If you need a string representation,
      * use getNormalizedIdentifier resp. getUrlSafeIdentifier
      *
-     * @param object $model
-     *
-     * @return array list of all identifiers of this model
+     * @return mixed[]
      */
-    public function getIdentifierValues($model);
+    public function getIdentifierValues(object $model): array;
 
     /**
-     * Get a list of the field names models of the specified class use to store
-     * the identifier.
+     * Get a list of the field names models of the specified fully qualified
+     * class name used to store the identifier.
      *
-     * @param string $class fully qualified class name
+     * @return string[]
      *
-     * @return array
+     * @phpstan-param class-string $class
      */
-    public function getIdentifierFieldNames($class);
+    public function getIdentifierFieldNames(string $class): array;
 
     /**
      * Get the identifiers for this model class as a string.
-     *
-     * @param object $model
-     *
-     * @return string a string representation of the identifiers for this
-     *                instance
      */
-    public function getNormalizedIdentifier($model);
+    public function getNormalizedIdentifier(object $model): string;
 
     /**
      * Get the identifiers as a string that is safe to use in a url.
      *
      * This is similar to getNormalizedIdentifier but guarantees an id that can
      * be used in a URL.
-     *
-     * @param object $model
-     *
-     * @return string string representation of the id that is safe to use in a url
      */
-    public function getUrlSafeIdentifier($model);
+    public function getUrlSafeIdentifier(object $model): string;
 
     /**
      * Create a new instance of the model of the specified class.
      *
-     * @param string $class
+     * @template T of object
      *
-     * @return object
+     * @phpstan-param class-string<T> $class
+     * @phpstan-return T
      */
-    public function getModelInstance($class);
+    public function getModelInstance(string $class): object;
 
     /**
-     * @param string $class
+     * @template T of object
+     *
+     * @return Collection<int|string, object>
      *
-     * @return array|\ArrayAccess
+     * @phpstan-param class-string<T> $class
+     * @phpstan-return Collection<array-key, T>
      */
-    public function getModelCollectionInstance($class);
+    public function getModelCollectionInstance(string $class): Collection;
```

Comment:
This method on 3.x was deprecated, (will be removed once we merge 3.x with master, and we removed deprecated code again) but I wanted to show how this would have been.

Reply:
I like this way.